File: vben-admin/apps/web-antd/src/api/platform-config.ts
import { requestClient } from '#/api/request';

export interface PlatformConfig {
  id: number;
  pt: string;
  name: string;
  auth_type: string;
  success_codes: string;
  use_json: boolean;
  need_proxy: boolean;
  returns_yid: boolean;
  extra_params: boolean;
  query_act: string;
  query_path: string;
  query_method: string;
  query_body_type: string;
  query_param_style: string;
  query_param_map: string;
  query_polling: boolean;
  query_max_attempts: number;
  query_interval: number;
  query_response_map: string;
  order_path: string;
  order_method: string;
  order_body_type: string;
  order_param_map: string;
  yid_in_data_array: boolean;
  progress_path: string;
  progress_method: string;
  progress_body_type: string;
  progress_param_map: string;
  batch_progress_path: string;
  batch_progress_method: string;
  batch_progress_body_type: string;
  batch_progress_param_map: string;
  category_path: string;
  category_method: string;
  category_body_type: string;
  category_param_map: string;
  class_list_path: string;
  class_list_method: string;
  class_list_body_type: string;
  class_list_param_map: string;
  pause_path: string;
  pause_method: string;
  pause_body_type: string;
  pause_param_map: string;
  pause_id_param: string;
  resume_path: string;
  resume_method: string;
  resume_body_type: string;
  resume_param_map: string;
  change_pass_path: string;
  change_pass_method: string;
  change_pass_body_type: string;
  change_pass_param_map: string;
  change_pass_param: string;
  change_pass_id_param: string;
  resubmit_path: string;
  resubmit_method: string;
  resubmit_body_type: string;
  resubmit_param_map: string;
  resubmit_id_param: string;
  log_path: string;
  log_method: string;
  log_body_type: string;
  log_param_map: string;
  log_id_param: string;
  balance_path: string;
  balance_money_field: string;
  balance_method: string;
  balance_body_type: string;
  balance_param_map: string;
  balance_auth_type: string;
  report_param_style: string;
  report_auth_type: string;
  report_path: string;
  report_method: string;
  report_body_type: string;
  report_param_map: string;
  get_report_path: string;
  get_report_method: string;
  get_report_body_type: string;
  get_report_param_map: string;
  refresh_path: string;
  source_code: string;
  created_at: string;
  updated_at: string;
}

export interface ParsedPHPResult {
  pt: string;
  name: string;
  auth_type: string;
  api_path_style: string;
  success_codes: string;
  use_json: boolean;
  query_act: string;
  query_path: string;
  order_act: string;
  order_path: string;
  progress_act: string;
  progress_path: string;
  progress_method: string;
  pause_act: string;
  pause_path: string;
  pause_id_param?: string;
  change_pass_act: string;
  change_pass_path: string;
  change_pass_param: string;
  change_pass_id_param: string;
  log_act: string;
  log_path: string;
  log_id_param: string;
  returns_yid: boolean;
  balance_act: string;
  balance_path: string;
  balance_money_field: string;
  confidence: number;
  warnings: string[];
}

/** 获取所有平台配置 */
export async function getPlatformConfigsApi() {
  return requestClient.get<PlatformConfig[]>('/admin/platform-configs');
}

/** 保存平台配置 */
export async function savePlatformConfigApi(data: Partial<PlatformConfig>) {
  return requestClient.post('/admin/platform-config/save', data);
}

/** 删除平台配置 */
export async function deletePlatformConfigApi(pt: string) {
  return requestClient.delete(`/admin/platform-config/${pt}`);
}

/** 解析 PHP 代码 */
export async function parsePHPCodeApi(code: string) {
  return requestClient.post<ParsedPHPResult>('/admin/platform-config/parse-php', { code });
}

export interface DetectRequest {
  url: string;
  uid: string;
  key: string;
  token: string;
  cookie: string;
}

export interface ProbeDetail {
  endpoint: string;
  method: string;
  status: string;
  code: string;
  msg: string;
  raw_body: string;
}

export interface DetectResult {
  success: boolean;
  auth_type: string;
  success_code: string;
  api_style: string;
  balance_ok: boolean;
  balance_money: string;
  balance_act: string;
  balance_path: string;
  balance_field: string;
  query_ok: boolean;
  query_act: string;
  class_list_ok: boolean;
  category_ok: boolean;
  use_json: boolean;
  returns_yid: boolean;
  probes: ProbeDetail[];
  suggested_name: string;
  config: Record<string, string>;
}

/** 自动检测平台 */
export async function detectPlatformApi(data: DetectRequest) {
  return requestClient.post<DetectResult>('/admin/platform-config/detect', data);
}

export interface AutoDetectSaveRequest extends DetectRequest {
  pt: string;
  name: string;
}

export interface AutoDetectSaveResult {
  success: boolean;
  msg: string;
  detect: DetectResult;
  config: Partial<PlatformConfig>;
}

/** 一键检测并保存平台配置 */
export async function autoDetectSaveApi(data: AutoDetectSaveRequest) {
  return requestClient.post<AutoDetectSaveResult>('/admin/platform-config/auto-detect-save', data);
}

File: go-api/internal/model/platform_config.go
package model

import "time"

// PlatformConfigDB 平台接口配置（数据库模型）
type PlatformConfigDB struct {
	ID                 int       `json:"id" db:"id"`
	PT                 string    `json:"pt" db:"pt"`
	Name               string    `json:"name" db:"name"`
	AuthType           string    `json:"auth_type" db:"auth_type"`
	SuccessCodes       string    `json:"success_codes" db:"success_codes"`
	UseJSON            bool      `json:"use_json" db:"use_json"`
	NeedProxy          bool      `json:"need_proxy" db:"need_proxy"`
	ReturnsYID         bool      `json:"returns_yid" db:"returns_yid"`
	ExtraParams        bool      `json:"extra_params" db:"extra_params"`
	QueryAct           string    `json:"query_act" db:"query_act"`
	QueryPath          string    `json:"query_path" db:"query_path"`
	QueryMethod        string    `json:"query_method" db:"query_method"`
	QueryBodyType      string    `json:"query_body_type" db:"query_body_type"`
	QueryParamStyle    string    `json:"query_param_style" db:"query_param_style"`
	QueryParamMap      string    `json:"query_param_map" db:"query_param_map"`
	QueryPolling       bool      `json:"query_polling" db:"query_polling"`
	QueryMaxAttempts   int       `json:"query_max_attempts" db:"query_max_attempts"`
	QueryInterval      int       `json:"query_interval" db:"query_interval"`
	QueryResponseMap   string    `json:"query_response_map" db:"query_response_map"`
	OrderPath          string    `json:"order_path" db:"order_path"`
	OrderMethod        string    `json:"order_method" db:"order_method"`
	OrderBodyType      string    `json:"order_body_type" db:"order_body_type"`
	OrderParamMap      string    `json:"order_param_map" db:"order_param_map"`
	YIDInDataArray     bool      `json:"yid_in_data_array" db:"yid_in_data_array"`
	ProgressPath       string    `json:"progress_path" db:"progress_path"`
	ProgressMethod     string    `json:"progress_method" db:"progress_method"`
	ProgressBodyType   string    `json:"progress_body_type" db:"progress_body_type"`
	ProgressParamMap   string    `json:"progress_param_map" db:"progress_param_map"`
	CategoryPath       string    `json:"category_path" db:"category_path"`
	CategoryMethod     string    `json:"category_method" db:"category_method"`
	CategoryBodyType   string    `json:"category_body_type" db:"category_body_type"`
	CategoryParamMap   string    `json:"category_param_map" db:"category_param_map"`
	ClassListPath      string    `json:"class_list_path" db:"class_list_path"`
	ClassListMethod    string    `json:"class_list_method" db:"class_list_method"`
	ClassListBodyType  string    `json:"class_list_body_type" db:"class_list_body_type"`
	ClassListParamMap  string    `json:"class_list_param_map" db:"class_list_param_map"`
	PausePath          string    `json:"pause_path" db:"pause_path"`
	PauseMethod        string    `json:"pause_method" db:"pause_method"`
	PauseBodyType      string    `json:"pause_body_type" db:"pause_body_type"`
	PauseParamMap      string    `json:"pause_param_map" db:"pause_param_map"`
	PauseIDParam       string    `json:"pause_id_param" db:"pause_id_param"`
	ResumePath         string    `json:"resume_path" db:"resume_path"`
	ResumeMethod       string    `json:"resume_method" db:"resume_method"`
	ResumeBodyType     string    `json:"resume_body_type" db:"resume_body_type"`
	ResumeParamMap     string    `json:"resume_param_map" db:"resume_param_map"`
	ChangePassPath     string    `json:"change_pass_path" db:"change_pass_path"`
	ChangePassMethod   string    `json:"change_pass_method" db:"change_pass_method"`
	ChangePassBodyType string    `json:"change_pass_body_type" db:"change_pass_body_type"`
	ChangePassParamMap string    `json:"change_pass_param_map" db:"change_pass_param_map"`
	ChangePassParam    string    `json:"change_pass_param" db:"change_pass_param"`
	ChangePassIDParam  string    `json:"change_pass_id_param" db:"change_pass_id_param"`
	ResubmitPath       string    `json:"resubmit_path" db:"resubmit_path"`
	ResubmitMethod     string    `json:"resubmit_method" db:"resubmit_method"`
	ResubmitBodyType   string    `json:"resubmit_body_type" db:"resubmit_body_type"`
	ResubmitParamMap   string    `json:"resubmit_param_map" db:"resubmit_param_map"`
	ResubmitIDParam    string    `json:"resubmit_id_param" db:"resubmit_id_param"`
	LogPath            string    `json:"log_path" db:"log_path"`
	LogMethod          string    `json:"log_method" db:"log_method"`
	LogBodyType        string    `json:"log_body_type" db:"log_body_type"`
	LogParamMap        string    `json:"log_param_map" db:"log_param_map"`
	LogIDParam         string    `json:"log_id_param" db:"log_id_param"`
	BalancePath        string    `json:"balance_path" db:"balance_path"`
	BalanceMoneyField  string    `json:"balance_money_field" db:"balance_money_field"`
	BalanceMethod      string    `json:"balance_method" db:"balance_method"`
	BalanceBodyType    string    `json:"balance_body_type" db:"balance_body_type"`
	BalanceParamMap    string    `json:"balance_param_map" db:"balance_param_map"`
	BalanceAuthType    string    `json:"balance_auth_type" db:"balance_auth_type"`
	ReportParamStyle   string    `json:"report_param_style" db:"report_param_style"`
	ReportAuthType     string    `json:"report_auth_type" db:"report_auth_type"`
	ReportMethod       string    `json:"report_method" db:"report_method"`
	ReportBodyType     string    `json:"report_body_type" db:"report_body_type"`
	ReportParamMap     string    `json:"report_param_map" db:"report_param_map"`
	ReportPath         string    `json:"report_path" db:"report_path"`
	GetReportMethod    string    `json:"get_report_method" db:"get_report_method"`
	GetReportBodyType  string    `json:"get_report_body_type" db:"get_report_body_type"`
	GetReportParamMap  string    `json:"get_report_param_map" db:"get_report_param_map"`
	GetReportPath      string    `json:"get_report_path" db:"get_report_path"`
	RefreshPath        string    `json:"refresh_path" db:"refresh_path"`
	SourceCode         string    `json:"source_code" db:"source_code"`
	CreatedAt          time.Time `json:"created_at" db:"created_at"`
	UpdatedAt          time.Time `json:"updated_at" db:"updated_at"`
}

// PlatformConfigSaveRequest 保存平台配置请求
type PlatformConfigSaveRequest struct {
	ID                 int    `json:"id"`
	PT                 string `json:"pt" binding:"required"`
	Name               string `json:"name"`
	AuthType           string `json:"auth_type"`
	SuccessCodes       string `json:"success_codes"`
	UseJSON            bool   `json:"use_json"`
	NeedProxy          bool   `json:"need_proxy"`
	ReturnsYID         bool   `json:"returns_yid"`
	ExtraParams        bool   `json:"extra_params"`
	QueryAct           string `json:"query_act"`
	QueryPath          string `json:"query_path"`
	QueryMethod        string `json:"query_method"`
	QueryBodyType      string `json:"query_body_type"`
	QueryParamStyle    string `json:"query_param_style"`
	QueryParamMap      string `json:"query_param_map"`
	QueryPolling       bool   `json:"query_polling"`
	QueryMaxAttempts   int    `json:"query_max_attempts"`
	QueryInterval      int    `json:"query_interval"`
	QueryResponseMap   string `json:"query_response_map"`
	OrderPath          string `json:"order_path"`
	OrderMethod        string `json:"order_method"`
	OrderBodyType      string `json:"order_body_type"`
	OrderParamMap      string `json:"order_param_map"`
	YIDInDataArray     bool   `json:"yid_in_data_array"`
	ProgressPath       string `json:"progress_path"`
	ProgressMethod     string `json:"progress_method"`
	ProgressBodyType   string `json:"progress_body_type"`
	ProgressParamMap   string `json:"progress_param_map"`
	CategoryPath       string `json:"category_path"`
	CategoryMethod     string `json:"category_method"`
	CategoryBodyType   string `json:"category_body_type"`
	CategoryParamMap   string `json:"category_param_map"`
	ClassListPath      string `json:"class_list_path"`
	ClassListMethod    string `json:"class_list_method"`
	ClassListBodyType  string `json:"class_list_body_type"`
	ClassListParamMap  string `json:"class_list_param_map"`
	PausePath          string `json:"pause_path"`
	PauseMethod        string `json:"pause_method"`
	PauseBodyType      string `json:"pause_body_type"`
	PauseParamMap      string `json:"pause_param_map"`
	PauseIDParam       string `json:"pause_id_param"`
	ResumePath         string `json:"resume_path"`
	ResumeMethod       string `json:"resume_method"`
	ResumeBodyType     string `json:"resume_body_type"`
	ResumeParamMap     string `json:"resume_param_map"`
	ChangePassPath     string `json:"change_pass_path"`
	ChangePassMethod   string `json:"change_pass_method"`
	ChangePassBodyType string `json:"change_pass_body_type"`
	ChangePassParamMap string `json:"change_pass_param_map"`
	ChangePassParam    string `json:"change_pass_param"`
	ChangePassIDParam  string `json:"change_pass_id_param"`
	ResubmitPath       string `json:"resubmit_path"`
	ResubmitMethod     string `json:"resubmit_method"`
	ResubmitBodyType   string `json:"resubmit_body_type"`
	ResubmitParamMap   string `json:"resubmit_param_map"`
	ResubmitIDParam    string `json:"resubmit_id_param"`
	LogPath            string `json:"log_path"`
	LogMethod          string `json:"log_method"`
	LogBodyType        string `json:"log_body_type"`
	LogParamMap        string `json:"log_param_map"`
	LogIDParam         string `json:"log_id_param"`
	BalancePath        string `json:"balance_path"`
	BalanceMoneyField  string `json:"balance_money_field"`
	BalanceMethod      string `json:"balance_method"`
	BalanceBodyType    string `json:"balance_body_type"`
	BalanceParamMap    string `json:"balance_param_map"`
	BalanceAuthType    string `json:"balance_auth_type"`
	ReportParamStyle   string `json:"report_param_style"`
	ReportAuthType     string `json:"report_auth_type"`
	ReportMethod       string `json:"report_method"`
	ReportBodyType     string `json:"report_body_type"`
	ReportParamMap     string `json:"report_param_map"`
	ReportPath         string `json:"report_path"`
	GetReportMethod    string `json:"get_report_method"`
	GetReportBodyType  string `json:"get_report_body_type"`
	GetReportParamMap  string `json:"get_report_param_map"`
	GetReportPath      string `json:"get_report_path"`
	RefreshPath        string `json:"refresh_path"`
	SourceCode         string `json:"source_code"`
}
